lib/rules.js

```diff
@@ -353,6 +353,267 @@ export const GROUP_TERMS_BY_ROOT = defineRule(
+        const coefficients = Object.keys(coefficientMap).map(key => {return coefficientMap[key]})
+
+        const firstCoef = query.getValue(coefficients[0][0])
+        const firstPoly = query.isMul(variables[0]) ? variables[0].args : [variables[0]]
+
+        const secondCoef = query.getValue(coefficients[1][0])
+
+        const thirdPoly =
+              variables[2]
+              ? query.isMul(variables[2])
+              ? variables[2].args
+              : [variables[2]]
+              : null
+
+        const thirdCoef =
+              thirdPoly ?
+              query.getValue(coefficients[2][0])
+              : constants[0]
+              ? query.getValue(constants[0])
+              : null
+
+        const firstFactors = getFactors(firstCoef)
+        const thirdFactors = getFactors(Math.abs(thirdCoef))
+
+        let combo
+
+        for (var i in firstFactors){
+            for (var j in thirdFactors) {
+                const l1 = firstFactors[i]
+                const l2 = thirdFactors[j]
+                // both positive
+                let one = l1[0] * l2[1] + l1[1] * l2[0]
+                let two = l1[0] * l2[0] + l1[1] * l2[1]
+                // both negative
+                let three = l1[0] * -l2[1] + l1[1] * -l2[0]
+                let four = l1[0] * -l2[0] + l1[1] * -l2[1]
+                // l2[0] is negative and l2[1] is positive
+                let five = l1[0] * l2[1] + l1[1] * -l2[0]
+                let six = l1[0] * -l2[0] + l1[1] * l2[1]
+                // l2[0] is positive and l2[1] is negative
+                let seven = l1[0] * -l2[1] + l1[1] * l2[0]
+                let eight = l1[0] * l2[0] + l1[1] * -l2[1]
+
+                if (one == secondCoef && l2[0] * l2[1] == thirdCoef){
+                    combo = [[l1[0],l2[0]],[l1[1],l2[1]]]
+                } else if (two == secondCoef && l2[0] * l2[1] == thirdCoef){
+                    combo = [[l1[0],l2[1]],[l1[1],l2[0]]]
+                } else if (three == secondCoef && -l2[0] * -l2[1] == thirdCoef){
+                    combo = [[l1[0],-l2[0]],[l1[1],-l2[1]]]
+                } else if (four == secondCoef && -l2[0] * -l2[1] == thirdCoef){
+                    combo = [[l1[0],-l2[1]],[l1[1],-l2[0]]]
+                } else if (five == secondCoef && -l2[0] * l2[1] == thirdCoef){
+                    combo = [[l1[0],-l2[0]],[l1[1],l2[1]]]
+                } else if (six == secondCoef && -l2[0] * l2[1] == thirdCoef){
+                    combo = [[l1[0],l2[1]],[l1[1],-l2[0]]]
+                } else if (seven == secondCoef && l2[0] * -l2[1] == thirdCoef){
+                    combo = [[l1[0],l2[0]],[l1[1],-l2[1]]]
+                } else if (eight == secondCoef && l2[0] * -l2[1] == thirdCoef){
+                    combo = [[l1[0],-l2[1]],[l1[1],l2[0]]]
+                } else {
+                    combo = []
+                }
+            }
+        }
+        if (combo == []) {
+            throw new Error('cannot factor polynomial')
+        }
```

Comment:
If feels weird that `canApplyRule` might be successful, but `applyRule` might still fail.